### PR TITLE
feat(gpu): optimize integer mul when one of the ct holds boolean value

### PR DESCRIPTION
**Pull request by bbarbakadze**
### PR content/description

Implements optimization for mul in the cuda backend, when one of the ciphertexts holds a boolean value.

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h [hunk 1]
```diff
@@ -1117,20 +1117,97 @@ template <typename Torus> struct int_sum_ciphertexts_vec_memory {
   }
 };
 
+template <typename Torus> struct int_zero_out_if_buffer {
```
Comment (agnesLeroy) [inline backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:1120]: Didn't we alreadyy have one like this? :thinking:

File: backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h [hunk 2]
```diff
@@ -1598,50 +1684,6 @@ template <typename Torus> struct int_arithmetic_scalar_shift_buffer {
   }
 };
 
-template <typename Torus> struct int_zero_out_if_buffer {
```
Comment (agnesLeroy) [inline backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:1601]: Ah it's just moved sorry it's ok :)

**Review by agnesLeroy — APPROVED** [review]

**Comment by agnesLeroy** [comment]
Thank you @bbarbakadze! Great job! :slightly_smiling_face:

File: backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h (hunk 1 above)
Comment (bbarbakadze) [reply backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:1120]: Yes I just moved it a bit above.

**Comment by agnesLeroy** [comment]
@aquint-zama the cla is stalled for some reason on this PR, do you know how we could trigger it?

**Pull request merged**